cmake/download_test_data.cmake

```diff
@@ -57,15 +57,21 @@ message(STATUS "Compiler: ${CXX_VERSION_RESULT}")
 
 # determine used C++ standard library (for debug and support purposes)
 if(NOT DEFINED LIBCPP_VERSION_OUTPUT_CACHED)
-    try_run(RUN_RESULT_VAR COMPILE_RESULT_VAR
-        "${CMAKE_BINARY_DIR}" SOURCES "${CMAKE_SOURCE_DIR}/cmake/detect_libcpp_version.cpp"
-        RUN_OUTPUT_VARIABLE LIBCPP_VERSION_OUTPUT COMPILE_OUTPUT_VARIABLE LIBCPP_VERSION_COMPILE_OUTPUT
-    )
-    if(NOT LIBCPP_VERSION_OUTPUT)
-        set(LIBCPP_VERSION_OUTPUT "Unknown")
-        message(AUTHOR_WARNING "Failed to compile cmake/detect_libcpp_version to detect the used C++ standard library. This does not affect the library or the test cases. Please still create an issue at https://github.com/nlohmann/json to investigate this.\n${LIBCPP_VERSION_COMPILE_OUTPUT}")
+    if(CMAKE_CROSSCOMPILING)
```

Comment:
No, I meant moving this in front of `if(NOT DEFINED LIBCPP_VERSION_OUTPUT_CACHED)` and defining `LIBCPP_VERSION_OUTPUT_CACHED` with "could not be detected due to cross-compiling". Then you can keep the code after `if(NOT DEFINED LIBCPP_VERSION_OUTPUT_CACHED)` as is, because `LIBCPP_VERSION_OUTPUT_CACHED` is defined.